### PR TITLE
Refactor/change query fields

### LINKED ISSUE #745: change the firebase query of assign task middleware
- Previously it was decided that we will store level and category of a task in a object named `taskLevel` as follows

```
taskLevel: {
  category: react,
  level: 2
}
```

but it makes more sense to store `category` and `level` separately on every tasks instead of putting it in an object,

- Our middleware currently makes a query based on the Object data type, we need to change that

### PR DESCRIPTION
**Pull request by vinayak-trivedi**
closes #745

### CODE REVIEW

File: middlewares/validators/tasks.js [hunk 1]
```diff
@@ -77,6 +84,13 @@ const updateTask = async (req, res, next) => {
           [NEELAM]: joi.number().optional(),
         })
         .optional(),
+      taskLevel: joi
+        .object()
+        .keys({
+          category: joi.string().optional(),
+          level: joi.number().optional(),
+        })
+        .optional(),
```
Comment (rohan09-raj) [inline middlewares/validators/tasks.js:93]: This should also be fixed as we are not using `taskLevel` object anymore, category and level should be individually validated.

Comment (vinayak-trivedi) [reply middlewares/validators/tasks.js:93]: I don't know why it's there, I pulled Muralidhar's PR some days back so I think when I created a new branch, those changes still were there, let me remove it

**Review by rohan09-raj — DISMISSED** [review]
LGTM 💯 💯

File: middlewares/assignTask.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ const tasks = firestore.collection("tasks");
 
 const assignTask = async function (req, res) {
   try {
-    const { task } = await fetchSkillLevelTask("frontend", 1);
+    const { task } = await fetchSkillLevelTask("FRONTEND", 1);
```
Comment (shubham-y) [inline middlewares/assignTask.js:7]: Can you please explain why are we passing skill `FRONTEND` by default?

Comment (shubham-y) [reply middlewares/assignTask.js:7]: Also, I think it will be better to import it to from constants

Comment (vinayak-trivedi) [reply middlewares/assignTask.js:7]: it is hard coded for now, it will come from the frontend when we have skills in the user object

Comment (shubham-y) [reply middlewares/assignTask.js:7]: Got it. As discussed, please add a comment in code for now

**Review by shubham-y — DISMISSED** [review]
LGTM 👍

File: middlewares/assignTask.js [hunk 3]
```diff
@@ -4,7 +4,8 @@ const tasks = firestore.collection("tasks");
 
 const assignTask = async function (req, res) {
   try {
-    const { task } = await fetchSkillLevelTask("frontend", 1);
+    // this hardcoded value will be removed once we have user skill
```
Comment (ankushdharkar) [inline middlewares/assignTask.js:7]: Should this be having a ticket @vinayak-trivedi?

File: models/tasks.js [hunk 4]
```diff
@@ -195,9 +195,9 @@ const fetchUserTasks = async (username, statuses = [], field, order) => {
 
 const getNewTask = async (skill, level) => {
   const task = await tasksModel
-    .where("taskLevel.category", "==", skill)
-    .where("taskLevel.level", ">=", level)
-    .where("taskLevel.level", "<=", level + 2)
+    .where("category", "==", skill)
+    .where("level", ">=", level)
+    .where("level", "<=", level + 2)
     .where("status", "==", "AVAILABLE")
```
Comment (ankushdharkar) [inline models/tasks.js:201]: These values should ideally come from a constants file

**Pull request merged**